lib/Sema/CSSimplify.cpp

```diff
@@ -4669,7 +4669,7 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
     ConstraintKind kind, Type baseTy, DeclName member, Type memberTy,
     DeclContext *useDC, FunctionRefKind functionRefKind,
     ArrayRef<OverloadChoice> outerAlternatives, TypeMatchOptions flags,
-    ConstraintLocatorBuilder locatorB) {
+    ConstraintLocatorBuilder locatorB, bool allowFixes) {
```

Comment:
I think the same effect could be archived with setting  `TypeMatchFlags::TMF_ApplyingFix` to `flags` instead of adding one more boolean parameter.

Reply:
Or locally setting `recordFixes` to `false` in `solveWithNewBaseOrName` via `llvm::SaveAndRestore<bool> (solverState->recordFixes, false);`